### ISSUE TITLE
[Bug] FlashAttention kernel error

### ISSUE THREAD

**Issue by hogura99** [USC-NSL/DisagMoE#14]
``` bash
Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/ut[0/1968]ice.py", line 79, in __torch_function__
(Engine pid=531663)     return func(*args, **kwargs)
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_ops.py", line 1061, in __call__
(Engine pid=531663)     return self_._op(*args, **(kwargs or {}))
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_library/autograd.py", line 98, in autograd_impl
(Engine pid=531663)     result = Generated.apply(*args, Metadata(keyset, keyword_only_args))  # type: ignore[attr-defined]
(Engine pid=531663)              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/autograd/function.py", line 574, in apply
(Engine pid=531663)     return super().apply(*args, **kwargs)  # type: ignore[misc]
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_library/autograd.py", line 40, in forward
(Engine pid=531663)     result = op.redispatch(keyset & _C._after_autograd_keyset, *args, **kwargs)
(Engine pid=531663)              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_ops.py", line 672, in redispatch
(Engine pid=531663)     return self_._handle.redispatch_boxed(keyset, *args, **kwargs)
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_library/custom_ops.py", line 236, in backend_impl
(Engine pid=531663)     result = self._backend_fns[device_type](*args, **kwargs)
(Engine pid=531663)              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/vllm/attention/backends/flash_attn.py", line 54, in flash_attn_varlen_func
(Engine pid=531663)     return _flash_attn_varlen_func(
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/vllm/vllm_flash_attn/flash_attn_interface.py", line 1155, in flash_attn_varlen_func
(Engine pid=531663)     return FlashAttnVarlenFunc.apply(
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/autograd/function.py", line 574, in apply
(Engine pid=531663)     return super().apply(*args, **kwargs)  # type: ignore[misc]
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/vllm/vllm_flash_attn/flash_attn_interface.py", line 633, in forward
(Engine pid=531663)     out, q, k, v, out_padded, softmax_lse, S_dmask, rng_state = _flash_attn_varlen_forward(
(Engine pid=531663)                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/vllm/vllm_flash_attn/flash_attn_interface.py", line 91, in _flash_attn_varlen_forward
(Engine pid=531663)     out, q, k, v, out_padded, softmax_lse, S_dmask, rng_state = torch.ops.vllm_flash_attn_c.varlen_fwd(
(Engine pid=531663)                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663)   File "/home/shaoyuw/miniconda3/envs/cu122/lib/python3.12/site-packages/torch/_ops.py", line 1061, in __call__
(Engine pid=531663)     return self_._op(*args, **(kwargs or {}))
(Engine pid=531663)            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Engine pid=531663) RuntimeError: CUDA error: an illegal memory access was encountered
```

**Comment by hogura99** [comment]
With `CUDA_LAUNCH_BLOCKING=1`

**Comment by hogura99** [comment]
Should have been solved in #19 .
